django/db/models/base.py

```diff
@@ -1087,12 +1087,31 @@ def _save_table(
             ]
             forced_update = update_fields or force_update
             pk_val = self._get_pk_val(meta)
-            updated = self._do_update(
-                base_qs, using, pk_val, values, update_fields, forced_update
+            returning_fields = [
+                f
+                for f in meta.local_concrete_fields
+                if f.generated
+                and f.referenced_fields.intersection(non_pks_non_generated)
+            ]
```

Comment:
This ensures only generated fields referencing one of the updated fields are requested to be returned.

If the model doesn't have any generated field or `save` is called with `update_fields` that no generated fields reference no returning fields will be requested.

Reply:
Maybe it's off-topic here, but, what if a generated field reference all fields with `db_default`?

```python
from django.db import models


class Item(models.Model):
    price = models.IntegerField(db_default=models.Value(1))
    quantity = models.IntegerField(db_default=models.Value(1))
    total_price = models.GeneratedField(
        expression=models.F("price") * models.F("quantity"),
        output_field=models.BigIntegerField(),
        db_persist=True,
    )
```

Reply:
@pauloxnet this MR is about adding support for `RETURNING` on `save` `UPDATE`. `db_default` is used on `save` `INSERT` for which we already retrieve `db_returning` fields

https://github.com/django/django/blob/dd133054cb98f77577c06d7ef1f2391a865784bc/django/db/models/base.py#L1122-L1125